### PR TITLE
added parsing where in expression

### PR DESCRIPTION
**Pull request by ir73**

### CODE REVIEW

File: src/main/java/com/flair/bi/compiler/snowflake/SnowflakeListener.java [hunk 1]
```diff
@@ -218,6 +218,10 @@ public void exitExpr(ExprContext ctx) {
                 .map(property::get)
                 .ifPresent(str::append);
 
+        if (str.length() == 0) {
```
Comment (admin-vizcentric) [inline src/main/java/com/flair/bi/compiler/snowflake/SnowflakeListener.java]: Use String#isEmpty() instead.

File: src/main/java/com/flair/bi/compiler/postgres/PostgresListener.java [hunk 2]
```diff
@@ -216,6 +216,10 @@ public void exitExpr(ExprContext ctx) {
                 .map(property::get)
                 .ifPresent(str::append);
 
+        if (str.length() == 0) {
```
Comment (admin-vizcentric) [inline src/main/java/com/flair/bi/compiler/postgres/PostgresListener.java]: Use String#isEmpty() instead.

File: src/main/java/com/flair/bi/compiler/oracle/OracleListener.java [hunk 3]
```diff
@@ -236,6 +236,10 @@ else if("YEARQUARTER".equalsIgnoreCase(ctx.func_call_expr().start.getText())) {
                 .map(property::get)
                 .ifPresent(str::append);
 
+		if (str.length() == 0) {
```
Comment (admin-vizcentric) [inline src/main/java/com/flair/bi/compiler/oracle/OracleListener.java]: Use String#isEmpty() instead.

File: src/main/java/com/flair/bi/compiler/mysql/MySQLListener.java [hunk 4]
```diff
@@ -199,6 +199,10 @@ else if(Optional.ofNullable(ctx.func_call_expr()).isPresent() && "YEARQUARTER".e
                 .map(property::get)
                 .ifPresent(str::append);
 
+        if (str.length() == 0) {
```
Comment (admin-vizcentric) [inline src/main/java/com/flair/bi/compiler/mysql/MySQLListener.java]: Use String#isEmpty() instead.

Comment (ir73) [reply src/main/java/com/flair/bi/compiler/mysql/MySQLListener.java]: StringBuilder does not have `isEmpty` method afaik.

**Pull request merged**